src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -92,6 +93,21 @@ public static Phone parsePhone(String phone) throws ParseException {
         return new Phone(trimmedPhone);
     }
 
+    /**
+     * Parses a {@code String username} into a {@code Telegram}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code username} is invalid.
+     */
+    public static Telegram parseTelegram(String username) throws ParseException {
+        requireNonNull(username);
+        String trimmedUsername = username.trim();
+        if (!Telegram.isValidTelegram(trimmedUsername) || trimmedUsername.equals("")) {
```

Comment:
I think isValidTelegram() already handles the case where the trimmedUsername equals "", so we can leave it out? Same for email and address.